.github/workflows/ci.yaml

```diff
@@ -0,0 +1,110 @@
+name: Run CI checks
+
+on: pull_request
+
+jobs:
+  run_lint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Install poetry
+        run: pipx install poetry
+      - uses: actions/cache@v3
+        id: cache-poetry
+        env:
+          SEGMENT_DOWNLOAD_TIMEOUT_MIN: "15"
+        with:
+          path: |
+            ~/.cache/pypoetry/virtualenvs
+            ~/.cache/pypoetry/cache
+            ~/.cache/pypoetry/artifacts
+          key: poetry-dev-${{ hashFiles('poetry.lock') }}
```

Comment:
We need to use this caching strategy (rather than setting `cache: poetry` in `setup-python`) because our use of dependency groups. We want separate caches for each dependency group.

https://github.com/actions/setup-python/issues/505

Reply:
(Going to include this as a comment)